frontend/src/pages/Metrics.jsx

```diff
@@ -0,0 +1,118 @@
+import React, { useEffect, useState } from 'react'
+import Layout from '../layouts/Layout.jsx'
+import { NavLink, useParams } from 'react-router-dom'
+import '../styles/pages/Metrics.css'
+import '@fortawesome/fontawesome-free/css/all.min.css';
+
+const MetricRaws = (props) => {
+    useEffect(() => {
+        const fetchMetrics = async () => {
+            const data = await props.getAllMetrics();
+            console.log("get data:: ", data);
+            props.setMetricData(data);
+            props.setLoading(false); 
+        };
+
+        fetchMetrics(); 
+    }, []);
+
+    if (props.loading) {
+        return <div>Loading...</div>; 
+    }
+
+    if (!props.metricData) {
+        return <div>No data available</div>;
+    }
+
+    return props.metricData.map((metric, index) => (
+        <tr key={index}>
```

Comment:
setting the key to the index is not the best thing to do, you already have the id of each metric, use it instead of the index.

Reply:
Done